### DISCUSSION TITLE
The instable communication between crazyradio and crazyflie

### DISCUSSION THREAD

**Discussion by powehi2020** [bitcraze/discussions#435] [General]
The communication between crazyradio and crazyflie is not stable，when I try to control cf indoors in Windows, it always lost connection. The firmware version of the  crazyradio is 99.55. Do you meet the problem like this ? How can I fix it?

**Comment by Jarvis-X** [comment]
Hi,

One thing you could try is to connect the crazyradio to a USB 3 (SS, preferably) port on your computer, or using a USB hub with external power supply, which will guarantee enough current to the radio, we had some problem in that department before.

Other than that, you can try connecting the cf with an external antenna (with u.FL connector).

Jiawei

**Reply by ataffanel** [reply]
This is an interesting experience: we have mostly experienced the opposite: USB3 ports makes the radio communication less reliable. The radio should not be able to pull more power than a USB2 bus can provide so as long at it is not connected to a busy self-powered hub I would not expect power supply to be a problem.

**Comment by ataffanel** [comment]
The radio communication works usually pretty well so it is not normal if it is dropping very often. A couple of things you may try:
 -  Change the channel to a less busy one: we share the 2.4GHz spectrum with Wifi and in some location the Wifi networks can be very strong and busy so experimenting with other channels can help
 - Do not use a USB3 port, use a USB2 hub or  USB2 expansion cable on a USB3 port: USB3 works at 5Gbit/s which is almost exactly 2 times 2.4GHz, it means that it can cause interference with 2.4GHz radios.
 - Make sure you are using 2Mbps bitrates: I have found 1Mbps and 250Kbps to be much less reliable indoor.
